test/unit/org/apache/cassandra/index/sai/cql/NativeIndexDDLTest.java

```diff
@@ -264,16 +265,16 @@ public void shouldNotFailCreateWithTupleType()
     }
 
     @Test
-    public void shouldFailCreateWithInvalidCharactersInColumnName()
+    public void reproFailOnLongIndexName()
     {
-        String invalidColumn = "/invalid";
-        createTable(String.format("CREATE TABLE %%s (id text PRIMARY KEY, \"%s\" text)", invalidColumn));
+        String longIndexName = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab";
+        createTable("CREATE TABLE %s (key int PRIMARY KEY, value1 int, value2 int)");
+        createIndex(String.format("CREATE CUSTOM INDEX %s ON %%s(value1) USING 'StorageAttachedIndex'", longIndexName));
+        execute("INSERT INTO %s (\"key\", value1) VALUES (1, 1)");
+        execute("INSERT INTO %s (\"key\", value2) VALUES (2, 2)");
 
-        assertThatThrownBy(() -> executeNet(String.format("CREATE CUSTOM INDEX ON %%s(\"%s\")" +
-                                                          " USING 'StorageAttachedIndex'", invalidColumn)))
-                .isInstanceOf(InvalidQueryException.class)
-                .hasMessage(String.format("Column '%s' is longer than the permissible name length of %d characters or" +
-                                          " contains non-alphanumeric-underscore characters", invalidColumn, SchemaConstants.NAME_LENGTH));
+        RuntimeException e = assertThrows(RuntimeException.class, this::flush);
```

Comment:
Why do we allow long index names if they are going to fail on flush later? Shouldn't we normalize those names when creating the file?

Reply:
This is outside the scope of this issue as it's not a simple fix to not break existing schemas. See https://github.com/riptano/cndb/issues/13198